src/core/jwt/jwtTokenHandler.js

```diff
@@ -102,8 +103,7 @@ const jwtTokenHandlerFactory = function jwtTokenHandlerFactory({
                 if (response.status === 200) {
                     return response.json();
                 }
-                const error = new Error('Unsuccessful token refresh');
-                error.response = response;
+                const error = new TokenError('Refresh-token expired', response);
```

Comment:
I think it's better to check `response.status == 401` and in other cases throw old `new Error('Unsuccessful token refresh')`? 

For me it returns 401 if token is expired: I changed to small `jwt_access_token_ttl`/`jwt_refresh_token_ttl` in `nextgen-stack\tao-deliver-be\config\services.yaml` and restarted containers. I think BE sets refresh-token cookie expiration time, and so expired cookie gets removed. Second time I also increased cookie expiration time in browser DevTools, to send cookie to backend, but response was still 401. I don't have the latest backend version, though.

Reply:
I created the situation where this handler get 500 error, so splitting this on two errors is reasonable.

Reply:
In my case token expiration error redirects to external page, so it made `TokenError` as unrecoverable. At the moment we have two cases for errors: 
1. if code 401 - unrecoverable error with redirection on external page
2. all other errors are still recoverable with showing error page

Reply:
I agree with this